release/cut-official-release.sh

```diff
@@ -0,0 +1,338 @@
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Sets DIR, INSTRUCTIONS
+function main() {
+  # Parse arguments
+  if [[ "$#" -ne 2 ]]; then
+    usage
+    exit 1
+  fi
+  local -r new_version=${1-}
+  # TODO(ihmccreery) Stop calling it githash; it's not a githash.
+  local -r githash=${2-}
+  DRY_RUN=true
+  if [[ "${3-}" == "--no-dry-run" ]]; then
+    echo "!!! This NOT is a dry run."
+    DRY_RUN=false
+  else
+    echo "This is a dry run."
+  fi
+
+  check-prereqs
+
+  # Get and verify version info
+  local -r alpha_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.0-alpha\\.([1-9][0-9]*)$"
+  local -r beta_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-beta\\.([1-9][0-9]*)$"
+  local -r official_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  local -r series_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  if [[ "${new_version}" =~ $alpha_version_regex ]]; then
+    local -r release_type='alpha'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_alpha_rev="${BASH_REMATCH[3]}"
+  elif [[ "${new_version}" =~ $beta_version_regex ]]; then
+    local -r release_type='beta'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_patch="${BASH_REMATCH[3]}"
+    local -r version_beta_rev="${BASH_REMATCH[4]}"
+  elif [[ "${new_version}" =~ $official_version_regex ]]; then
+    local -r release_type='official'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_patch="${BASH_REMATCH[3]}"
+  elif [[ "${new_version}" =~ $series_version_regex ]]; then
+    local -r release_type='series'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+  else
+    usage
+    echo
+    echo "!!! You specified an invalid version '${new_version}'."
+    exit 1
+  fi
+
+  # Get the git commit from the githash and verify it
+  local -r git_commit_regex="^[0-9a-f]{7}$"
+  local -r git_commit=$(echo "${githash}" | awk -F'+' '{print $2}' | head -c7)
+  if ! [[ "${git_commit}" =~ $git_commit_regex ]]; then
+    usage
+    echo
+    echo "!!! You specified an invalid githash '${githash}'."
+    echo "!!! Tried to extract commit, got ${git_commit}."
+    exit 1
+  fi
+  echo "Doing ${release_type} release '${new_version}'."
+
+  # Set the default umask for the release. This ensures consistency
+  # across our release builds.
+  #
+  # TODO(ihmccreery): This should be in our build process, not our release
+  # process.
+  local -r release_umask=${release_umask:-022}
+  umask "${release_umask}"
+
+  # Start a tmp file that will hold instructions for the user.
+  declare -r INSTRUCTIONS="/tmp/kubernetes-${release_type}-release-${new_version}-$(date +%s)-instructions"
+  if $DRY_RUN; then
+    cat > "${INSTRUCTIONS}" <<- EOM
```

Comment:
This seems like gratuitous use of this syntax.  Would two echo's not be clearer?